src/harness/sas_testcase.py

```diff
@@ -142,16 +142,16 @@ def TriggerDailyActivitiesImmediatelyAndWaitUntilComplete(self):
     signal.alarm(0)
     
   def assertChannelsContainFrequencyRange(self, channels, frequency_range):
-    channels.sort(key=lambda ch: (ch['frequencyRange']['lowFrequency'], \
-                                  ch['frequencyRange']['highFrequency']), reverse = False)
+    channels.sort(key=lambda ch: (ch['frequencyRange']['lowFrequency']), reverse = False)
```

Comment:
You should keep the previous sort (sorting by freq_low but also by freq_high when same freq_low).
With this new sort, tests will fail for example with:

      channels = [{'frequencyRange' : {'lowFrequency': 3550, 'highFrequency': 3565}},
                          {'frequencyRange' : {'lowFrequency': 3550, 'highFrequency': 3560}}, 
                          {'frequencyRange' : {'lowFrequency': 3565, 'highFrequency': 3575}}]

because the new sort has no effect.
And the lowFrequency of third channel is not Less or Equal to highFrequency of second channel (line 151 assert).

Reply:
thank you for this comment I corrected it please review the changes.